### PR TITLE
chore: add test workflow for opentrons-ai-client

### PR DESCRIPTION
**Pull request by koji**
# Overview
add test workflow for opentrons-ai-client

solve AUTH-210 partially

[the result]
https://github.com/Opentrons/opentrons/actions/runs/8711151730/job/23894628978?pr=14924

# Test Plan

# Changelog

# Review requests

# Risk assessment
low

### CODE REVIEW

File: .github/workflows/opentrons-ai-client-test-build-deploy.yaml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      - '*.js'
+      - '*.json'
+      - 'yarn.lock'
+      - '.github/workflows/app-test-build-deploy.yaml'
+      - '.github/workflows/utils.js'
+    branches:
+      - '**'
+    tags:
+      - 'v*'
+      - 'ot3@*'
```
Comment (Elyorcv) [inline .github/workflows/opentrons-ai-client-test-build-deploy.yaml:22]: ot3 means `flex`?

**Review by jerader — APPROVED** [review]
github workflows are out of my element but the check passes so thats good 😅

File: .github/workflows/opentrons-ai-client-test-build-deploy.yaml (hunk 1 above)
Comment (koji) [reply .github/workflows/opentrons-ai-client-test-build-deploy.yaml:22]: yes
we keep using ot3 for some configurations and tags.

**Comment by koji** [comment]
I will update this workflow later to add build and deploy workflows.

**Pull request merged**